python/sglang/srt/layers/communicator.py

```diff
@@ -397,8 +399,19 @@ def _gather_hidden_states_and_residual(
             if hidden_states.shape[0] != 0:
                 hidden_states = layernorm(hidden_states)
         else:
-            hidden_states = tensor_model_parallel_all_reduce(hidden_states)
-            hidden_states, residual = layernorm(hidden_states, residual)
+            if (
+                is_sm100_supported()
+                and is_flashinfer_available()
```

Comment:
same as above

Reply:
Done in https://github.com/sgl-project/sglang/pull/7621/commits/cab45c3d1a66f26b15e41a2f2178fa52a576957d